### ISSUE TITLE
Bug: CC0033 appears again after adding 'this' keyword

### ISSUE THREAD

**Issue by ElStefan** [code-cracker/code-cracker#795]
Hi, I think I found a bug in the analyzer for CC0033

CC0033 is set to throw a warning, if a disposable field should be disposed.

Before the fix it looks like this:
````csharp
public class C : IDisposable
{
    private readonly MemoryStream ms = new MemoryStream();
    public void Dispose()
    {
        GC.SuppressFinalize(this);
    }
}    
````

After fix applied:
````csharp
public class C : IDisposable
{
    private readonly MemoryStream ms = new MemoryStream();
    public void Dispose()
    {
        GC.SuppressFinalize(this);
        ms.Dispose();
    }
}
````

According to our coding conventions, we add `this` in front of every private field, so we end up like this:
````csharp
public class C : IDisposable
{
    private readonly MemoryStream ms = new MemoryStream();
    public void Dispose()
    {
        GC.SuppressFinalize(this);
        this.ms.Dispose();
    }
}
````

But after that, CC0033 appears again, which is a little bit annoying.

The PR should go into the `v1.0.x` branch.

**Comment by giggio** [comment]
Bug confirmed.
@ElStefan I have updated the code to be easier to repro, and added some extra info. Thanks for the bug report.

**Comment by giggio** [comment]
Closed on #818
